packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -761,8 +911,16 @@ class MemoryController extends DisposableController
     snapshotByLibraryData ??= lastSnapshot?.librariesToList();
   }
 
-  void storeSnapshot(DateTime timestamp, HeapSnapshotGraph graph) {
-    snapshots.add(Snapshot(timestamp, this, graph));
+  void storeSnapshot(
+    DateTime timestamp,
+    HeapSnapshotGraph graph,
+    LibraryReference libraryRoot,
+  ) {
+    snapshots.value.add(Snapshot(timestamp, this, graph, libraryRoot));
```

Comment:
this won't cause the valueNotifier to actually notify. You would instead need to write:
`snapshots.value = snapshot.value.toList().add(..)`

Reply:
Originally, in old code, was current snapshot a ValueNotifier made snapshot snapshots but forgot to the ValueNotifiert.  Just a list.